### ISSUE TITLE
Web applications don't load on Windows Edge or Internet Explorer

### ISSUE THREAD

**Issue by Constans** [flutter/flutter#49786]
## Steps to Reproduce

1. Open either of the 2 browsers (Edge or IE)
2. Go to https://flutter.github.io/samples/
3. Nothing loads

**Expected results:** 

The app loads or a default page showing the currently supported browsers & versions is displayed.

**Actual results:** 

For the samples a black page displays / nothing happens.

For other apps:

A blank page with the **This page can't be displayed** or **Hmm... Can't reach this page**  titles show or worse it suggests that the page is unsafe and recommends to enable lower TLS support in the browser althought the browser has those settings on.

https://adobe.ly/2Oc85l9

<details>
  <summary>Flutter details</summary>

```
Flutter 1.14.7-pre.38 • channel master • https://github.com/flutter/flutter.git
Framework • revision 92f7e16312 (10 hours ago) • 2020-01-29 17:51:31 -0800
Engine • revision 6007c17fd2
Tools • Dart 2.8.0 (build 2.8.0-dev.5.0 fc3af737c7)
```

</details>

**Comment by VladyslavBondarenko** [comment]
Related to https://github.com/flutter/flutter/issues/45162

**Comment by Constans** [comment]
I've tested today with latest version of Flutter from master channel:

```
Flutter 1.15.2-pre.37 • channel master • https://github.com/flutter/flutter.git
Framework • revision b953c3e5a8 (54 minutes ago) • 2020-02-07 16:35:55 +0100
Engine • revision f34bc65bee
Tools • Dart 2.8.0 (build 2.8.0-dev.7.0 b3396cbdca)

Running flutter doctor...
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.15.2-pre.37, on Mac OS X 10.15.2 19C57, locale en-NL)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.5)
[✓] IntelliJ IDEA Ultimate Edition (version 2017.1.6)
[✓] Connected device (2 available)

• No issues found!
```

I've created a blank flutter project (with small edits like setting the brightness to dark and removing the debug checker) and it seems to work on Microsoft Edge:
https://constanting.com/flutter/tests/202002071738/index.html

This doesn't work on Internet Explorer 11.592.18362.0

The problem is that a previously created project, compiled with the same Flutter version doesn't work.

The structure is a bit different and there are some files missing with the old one - manifest.json, favicon.png, icons folder (these are added by default when creating a new Flutter project at this point).

I've tried to enhance the old index.html to be similar to the new one that's getting generated, but with no luck.

Also run **flutter clean** then **flutter build web** no luck as well.

So the issues might be around these differences between an old project and a new one.

The dev& compiling is done from VSCode latest version on what is updateable.

**Comment by Constans** [comment]
I made another test with the latest version from the **master** channel:
https://constanting.com/flutter/tests/202002271307/index.html
This demo project works fine, though the old ones still don't display a thing.

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.15.4-pre.192, on Mac OS X 10.15.2 19C57,
    locale en-NL)
 
[✓] Android toolchain - develop for Android devices (Android SDK version
    28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.5)
[✓] IntelliJ IDEA Ultimate Edition (version 2017.1.6)
[✓] Connected device (2 available)

• No issues found!
```

I get this error in the console from JS on Edge for the projects that are not working:

```
SCRIPT16389: Unspecified error. 
main.dart.js (29326,1)
```

And at that line there is this function:
```
P.afI.prototype={
$0:function(){var t,s=this.a,r=s.a
s=r==null?s.a= new P.fm():r
r=this.b
if(r==null)throw H.d(s)
t=H.d(s)
t.stack=r.j(0)
throw t),
$S:0}
```

Maybe this helps somehow.

**Comment by Constans** [comment]
Small update:

On Microsoft Edge, the same error gets thrown on the https://flutter.github.io/samples/#/ at line 44469:

```
P.aBC.prototype={
$0:function(){var t,s=this.a,r=s.a
s=r==null?s.a=new P.iW():r
r=this.b
if(r==null)throw H.m(s)
t=H.m(s)
t.stack=r.i(0)
throw t),
$S:1}
```
The difference is that the samples works on Edge and IE. There are significant differences betwen the HTML templates between the projects I have issues with and the ones that are present in the samples app.

I wonder also what version of Flutter was used to compile the samples?

**Comment by ferhatb** [comment]
Flutter Gallery now working with both on master. If you see failures, please file a new issue with console stack trace with a --profile build and if possible repro snippet.

**Comment by Constans** [comment]
Thank you! 
The gallery becomes unresponsive on IE (not responding due to a long-running script), and works Edge. 
My apps don't work yet, and that's still a mystery. 
I shall investigate further and post the resolution here just for the record.

**Comment by ferhatb** [comment]
Hi @Constans, do you see anything in console with your app if you run using :  flutter run with --profile ? Also, try turning on stop on all exceptions do get accurate stack trace. You'll see TextDecoder exception that is handled, if you continue execution after that you should see an exception that points to your app specifically. Please file a new issue with that stack trace, hope we can unblock you quickly with this.

**Comment by Constans** [comment]
Hello @ferhatb, if I run the apps from IDE (VS Code) it runs in Chrome. 
These apps don't have issues running on Chrome. 

They don't run on Edge/IE.

Is there a way to profile / debug in Edge directly on a Windows machine?

Also I got this error when running local web-server in Edge:

`Could not locate org-dartlang-sdk:///sdk/lib/_internal/js_runtime/lib/async_patch.dart specified in source map http://localhost:57905/main.dart.js.map.`

and a more verbose version of the previous mentioned error:

```
P._rootHandleUncaughtError_closure.prototype = {
    call$0: function() {
      var error,
        t1 = this._box_0,
        t2 = t1.stackTrace;
      if (t2 == null)
        throw H.wrapException(t1.error);
      error = H.wrapException(t1.error);
      error.stack = t2.toString$0(0);
      throw error;
    },
    $signature: 0
  };
```

Another weird thing on local is that when I open the Network introspection seems that app tries to load all dart files of the entire Flutter framework (with a 404 response). Maybe a map gone wild. 

All these results are with 1.16.0 version on the **dev** channel.

I'll dig more and might open a new issue if I still don't get to the bottom of it.

**Comment by ferhatb** [comment]
That is the root exception handler. If you set your debugger on Edge to stop on all exceptions, you'll see the original exception. Looking at that stack, going back a few elements, should show location of crash.

**Comment by Constans** [comment]
@ferhatb, thanks for following up with me!
The stack says something like: 

```
stack   
NoSuchMethodError: 
Object doesn't support property or method 'getAllKeys' (Error 438)
at H.wrapException (http://localhost:61250/main.dart.js:3775:7) 
at P._rootHandleUncaughtError_closure.prototype.call$0 (http://localhost:61250/main.dart.js:54020:7) 
at P._microtaskLoop (http://localhost:61250/main.dart.js:7106:9) 
at P._startMicrotaskLoop (http://localhost:61250/main.dart.js:7112:9) 
at P._AsyncRun__initializeScheduleImmediate_internalCallback.prototype.call$1 (http://localhost:61250/main.dart.js:51596:7) 
at H.invokeClosure (http://localhost:61250/main.dart.js:3971:11)
at Anonymous function (http://localhost:61250/main.dart.js:3990:11)
```

Which doesn't tell me much.
I'll keep digging. 

On Chrome running with debugging threw an error that I fixed, though that was unrelated with this one that happens on Edge/IE.
